app/controllers/cart_products_controller.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class CartProductsController < ApplicationController
+  before_action :validate_items_number
+
+  def create
+    cart_product = CartProduct.create(cart_id: cart_product_params[:cart_id], product_id: cart_product_params[:product_id], counter: cart_product_params[:number_to_add])
+
+    redirect_to cart_product.cart
+  end
+
+  def update
+    cart_product = CartProduct.find(params[:id])
+    counter = cart_product.counter + cart_product_params[:number_to_add].to_i
```

Comment:
wynioslbym to do SO, poniewaz tak bedzie Ci prosciej to przetestowac - request specka nie jest dotego dobrym rodzajem testu